### DISCUSSION TITLE
VW needs some kind of workaround for Organization invites without an email setup

### DISCUSSION THREAD

**Discussion by Leonetienne** [dani-garcia/vaultwarden#2856] [Ideas]
I am currently having the problem of my cloud provider completely blocking the ports 25 and 465, because some people abused it. They are now only unlocking them after specifically applying for it.
Thus, I can't get the mailing service to run without a major hassle.

I got the VW accounts email-verified, by manually setting the "verified_at" timestamp in the database, but I am completely unable to invite them to any organization, because invitation works by email-invite only. Because of crypto-technical reasons these cannot be just activated in the database.

Orgas are a pretty major feature and its sad to see that they are basically unusable and broken if there's no working email setup.
It would be great to see some kind of workaround, and if it's just some kind of log to retrieve the invitation link from the vaultwarden container...

What do you think?

**Comment by BlackDex** [comment]
You can just disable SMTP, that would solve your problem.
If you can't email at all, then it isn't useful to have it enabled in the first place.

**Reply by Leonetienne** [reply]
That does? Would it then just display the invitation link?
I mean, I tried, but without initiating a new invite. I just tried to resend the existing invite. That gave me an error, saying that smtp is not configured.

**Reply by BlackDex** [reply]
With SMTP off, you can invite, the  that user can just create a new account using that mail address. All accounts invited before disabling SMTP need to be deleted via the admin interface and invited again, because now it expects a valid token.

**Reply by BlackDex** [reply]
Btw, they only need to be deleted if they aren't active yet of course, only the pending invites need to be re-created

**Reply by Leonetienne** [reply]
Oh, I was talking about sharing passwords in organizations. I managed to create the users, and manually "activated" them via an sql query.
But this post is about organizations, which you can use to share passwords with friends or colleagues. These are invite-only, and an invitation will require some kind of key-setup with the invitee, which is why this can't be done manually in the DB. At this point multiple accounts have to be able to decrypt the same records, after all.

For me, this is is resolved. I applied for these ports to be opened, and some-freaking-how my request got accepted one hour afterwards. It is 2 AM on a Saturday where I and my provider live.

But this might still be relevant to others who can't setup emails for some reason.

**Reply by BlackDex** [reply]
I was talking about the same.

Vaultwarden can be used without any SMTP server. You can create accounts, invite people to organizations or create an emergency contact, all without ever sending one mail and manually editing the database. But for that you need to disable SMTP. That's all.

**Reply by tomholford** [reply]
**edit**: please disregard the below; for any future searchers, please refer to this section of the docs:

https://github.com/dani-garcia/vaultwarden/wiki/Differences-from-the-upstream-API-implementation#without-smtp-enabled

~@BlackDex I am working through this issue now with my org's instance.~

~In our case, one of our users had already created an account via the registration form. Then, via the Org admin UI, I invited him to join. Then, I noticed the "Needs confirmation" UI elements, did some searching and found this thread. So I disabled SMTP (which was enabled by default), removed him from and org and tried re-inviting. He still is unable to join - does his account need to be deleted and recreated due to the below? (thanks for your assistance)~

>~With SMTP off, you can invite, the that user can just create a new account using that mail address. All accounts invited before disabling SMTP need to be deleted via the admin interface and invited again, because now it expects a valid token.~
